current/runtime/storage.md

```diff
@@ -4,49 +4,237 @@ lang: en
+  [`IterableStorageMap#iter()`](https://substrate.dev/rustdocs/master/frame_support/storage/trait.IterableStorageMap.html#tymethod.iter),
+  [`IterableStorageDoubleMap#iter(key1)`](https://substrate.dev/rustdocs/master/frame_support/storage/trait.IterableStorageDoubleMap.html#tymethod.iter)
+- `drain` - Remove all elements from the map and iterate through them in no particular order. If you
+  add elements to the map while doing this, you'll get undefined results. Docs:
+  [`IterableStorageMap#drain()`](https://substrate.dev/rustdocs/master/frame_support/storage/trait.IterableStorageMap.html#tymethod.drain),
+  [`IterableStorageDoubleMap#drain(key1)`](https://substrate.dev/rustdocs/master/frame_support/storage/trait.IterableStorageDoubleMap.html#tymethod.drain)
+- `translate` - Use the provided function to translate all elements of the map, in no particular
+  order. To remove an element from the map, return `None` from the translation function. Docs:
+  [`IterableStorageMap#translate(fn)`](https://substrate.dev/rustdocs/master/frame_support/storage/trait.IterableStorageMap.html#tymethod.translate),
+  [`IterableStorageDoubleMap#translate(fn)`](https://substrate.dev/rustdocs/master/frame_support/storage/trait.IterableStorageDoubleMap.html#tymethod.translate)
+
+#### Hashing Algorithms
+
+As mentioned above, a novel feature of Substrate Storage Maps is that they allow developers to
+specify the hashing algorithm that will be used when generating a map's keys. A Rust object that is
+used to encapsulate hashing logic is referred to as a "hasher". Broadly speaking, the hashers that
+are available to Substrate developers can be described in two ways: whether or not they are
+cryptographic and whether or not they produce output that is transparent.
+
+##### Cryptographic Hashing Algorithms
+
+Cryptographic hashing algorithms are those that use cryptography to make it challenging to use the
+input to the hashing algorithm to influence its output. For example, a cryptographic hashing
+algorithm would produce a wide distribution of outputs even if the inputs were the numbers 1
+through 10. It is critical to use cryptographic hashing algorithms when users are able to influence
+the keys of a Storage Map. Failure to do so creates an attack vector that makes it easy for
+malicious actors to degrade the performance of your blockchain network. An example of a map that
+should use a cryptographic hash algorithm to generate its keys is a map used to track account
+balances. In this case, it is important to use a cryptographic hashing algorithm so that an attacker
+cannot bombard your system with many small transfers to sequential account numbers; without a
+cryptographic hash algorithm this would create an imbalanced storage structure that would suffer in
+performance. Cryptographic hashing algorithms are more complex and resource-intensive than their
+non-cryptographic counterparts, which is why Substrate allows developers to select when they are
+used.
```

Comment:
would be good to name the most common hashes here and say which is secure and which is not.